File: Java/marketplace/CreateSplit.java
package examples.marketplace;

import me.pagar.api.PagarmeCoreApiClient;
import me.pagar.api.controllers.OrdersController;
import me.pagar.api.http.client.*;
import me.pagar.api.models.*;

import java.util.*;

public class CreateSplit {

    public static void main(String[] args) {

        // Configuration parameters and credentials
        String basicAuthUserName = "basicAuthUserName"; // The username to use with basic authentication
        String basicAuthPassword = "basicAuthPassword"; // The password to use with basic authentication

        PagarmeCoreApiClient client = new PagarmeCoreApiClient(basicAuthUserName, basicAuthPassword);

        OrdersController orders_controller = new OrdersController();

        CreateCustomerRequest customer = new CreateCustomerRequest();
        customer.setName("sdk customer order");
        customer.setEmail("tonystark@avengers.com");

        CreateCreditCardPaymentRequest credit_card_payment_request = new CreateCreditCardPaymentRequest();
        credit_card_payment_request.setCapture(true);
        credit_card_payment_request.setInstallments(2);
        credit_card_payment_request.setStatementDescriptor("description");

        credit_card_payment_request.setCard(new CreateCardRequest());
        credit_card_payment_request.getCard().setNumber("4000000000000010");
        credit_card_payment_request.getCard().setHolderName("Tony Stark");
        credit_card_payment_request.getCard().setExpMonth(1);
        credit_card_payment_request.getCard().setExpYear(2025);
        credit_card_payment_request.getCard().setCvv("123");

        CreateOrderRequest request = new CreateOrderRequest();

        CreateOrderItemRequest orderItem = new CreateOrderItemRequest();
        orderItem.setDescription("Tesseract Bracelet");
        orderItem.setQuantity(1);
        orderItem.setAmount(200000);
        ArrayList<CreateOrderItemRequest> listOrderItem = new ArrayList<CreateOrderItemRequest>();
        listOrderItem.add(orderItem);
        request.setItems(listOrderItem);

        CreatePaymentRequest paymentItem = new CreatePaymentRequest();
        paymentItem.setPaymentMethod("credit_card");
        paymentItem.setCreditCard(credit_card_payment_request);
        request.setCustomer(customer);
        ArrayList<CreatePaymentRequest> lisPaymentItem = new ArrayList<CreatePaymentRequest>();
        lisPaymentItem.add(paymentItem);
        request.setPayments(lisPaymentItem);

        CreateSplitRequest splitItemOne = new CreateSplitRequest();
        splitItemOne.setRecipientId("rp_L4kwWE5FDuNdXYQP");
        splitItemOne.setAmount(100000);
        splitItemOne.setType("flat");
        CreateSplitRequest splitItemTwo = new CreateSplitRequest();
        splitItemTwo.setRecipientId("rp_4jl0ra2h3bI8VBvR");
        splitItemTwo.setAmount(100000);
        splitItemTwo.setType("flat");

        List<CreateSplitRequest> listSplit = new ArrayList<CreateSplitRequest>();
        listSplit.add(splitItemOne);
        listSplit.add(splitItemTwo);
        paymentItem.setSplit(listSplit);

        orders_controller.createOrderAsync(request, null, new APICallBack<GetOrderResponse>() {
            @Override
            public void onSuccess(HttpContext context, GetOrderResponse response) {
                System.out.println("Split create!");
                System.out.println("Order ID: " + response.getId());

                for (GetChargeResponse chargeCreate : response.getCharges()) {
                    System.out.println("Charge ID: " + chargeCreate.getId());
                }

                System.out.println("Order Status: " + response.getStatus());

            }

            @Override
            public void onFailure(HttpContext context, Throwable error) {

                System.out.println("Status response: " + context.getResponse().getStatusCode());
                System.out.println(error.getMessage());
                error.printStackTrace();

            }
        });

    }
}

File: Python/marketplace/create_split.py
from pagarmecoreapi.pagarmecoreapi_client import PagarmecoreapiClient
from pagarmecoreapi.models import *
from pagarmecoreapi.controllers import *
from pagarmecoreapi.exceptions.error_exception import *

# Configuration parameters and credentials
basic_auth_user_name = 'basic_auth_user_name' # The username to use with basic authentication
basic_auth_password = 'basic_auth_password' # The password to use with basic authentication

PagarmecoreapiClient(basic_auth_user_name, basic_auth_password)

orders_controller = orders_controller.OrdersController()

customer = create_customer_request.CreateCustomerRequest()
customer.name = "sdk customer order"
customer.email = "tonystark@avengers.com"

credit_card = create_credit_card_payment_request.CreateCreditCardPaymentRequest()
credit_card.capture = True
credit_card.installments = 2
credit_card.statement_descriptor = "descriptor"
credit_card.card = create_card_request.CreateCardRequest()
credit_card.card.number = "4000000000000010"
credit_card.card.holder_name = "Tony Stark"
credit_card.card.exp_month = 1
credit_card.card.exp_year = 2025
credit_card.card.cvv = "123"

request = create_order_request.CreateOrderRequest()

request.items = [create_order_item_request.CreateOrderItemRequest()]
request.items[0].description = "Tesseract Bracelet"
request.items[0].quantity = 1
request.items[0].amount = 200000

request.payments = [create_payment_request.CreatePaymentRequest()]
request.payments[0].payment_method = "credit_card"
request.payments[0].credit_card = credit_card
request.customer = customer

request.payments[0].split = [create_split_request.CreateSplitRequest(),
                             create_split_request.CreateSplitRequest()]

request.payments[0].split[0].recipient_id = "rp_L4kwWE5FDuNdXYQP"
request.payments[0].split[0].amount = 100000
request.payments[0].split[0].mtype = "flat"

request.payments[0].split[1].recipient_id = "rp_4jl0ra2h3bI8VBvR"
request.payments[0].split[1].amount = 100000
request.payments[0].split[1].mtype = "flat"

try:
    result = orders_controller.create_order(request)
    assert result.status == "paid"
    print("Order id: ", result.id)
    print("Charge id: ", result.charges[0].id)
    print("Order result status: ", result.status)
except ErrorException as ex:
    print(ex.message)
    print("Errors: ", ex.errors)
except Exception as ex:
    raise ex
    